src/expressions/functions/builtInFunctions.string.ts

```diff
@@ -405,6 +405,33 @@ const fnNormalizeSpace: FunctionDefinitionType = function(
 	return sequenceFactory.singleton(createAtomicValue(string.replace(/\s+/g, ' '), 'xs:string'));
 };
 
+const fnTranslate: FunctionDefinitionType = (
```

Comment:
Could you try to use more descriptive names for arguments and variables? You can often take the names used in the spec description for inspiration. e.g.
`arg1` = `argSequence`
`arg2` = `mapStringSequence`
`arg3` = `transStringSequence`

`str1` = `arg`
`str2` = `mapString`
`str3` = `transString`

`e` = `character` or `c` as `c` is a commonly used as short for `character`